cocos/gfx/webgpu/webgpu-command-buffer.ts

```diff
@@ -0,0 +1,736 @@
+import { INT_MAX } from '../../core/math/bits';
+import { GeneralBarrier } from '../base/states/general-barrier';
+import { TextureBarrier } from '../base/states/texture-barrier';
+import { BufferBarrier } from '../base/states/buffer-barrier';
+import { DescUpdateFrequency, WebGPUDeviceManager } from './define';
+import { WebGPUSwapchain } from './webgpu-swapchain';
+import { WebGPUPipelineLayout } from './webgpu-pipeline-layout';
+import { WebGPUDescriptorSetLayout } from './webgpu-descriptor-set-layout';
+import { error } from '../../core';
+
+export interface IWebGPUDepthBias {
+    constantFactor: number;
+    clamp: number;
+    slopeFactor: number;
+}
+
+export interface IWebGPUDepthBounds {
+    minBounds: number;
+    maxBounds: number;
+}
+
+export interface IWebGPUStencilWriteMask {
+    face: StencilFace;
+    writeMask: number;
+}
+
+export interface IWebGPUStencilCompareMask {
+    face: StencilFace;
+    reference: number;
+    compareMask: number;
+}
+
+interface CommandEncoder { commandEncoder: GPUCommandEncoder, renderPassEncoder: GPURenderPassEncoder }
+let currPipelineState: WebGPUPipelineState | null = null;
+const descriptorSets: WebGPUDescriptorSet[] = [];
+const renderAreas: Rect[] = [];
+export class WebGPUCommandBuffer extends CommandBuffer {
+    public pipelineBarrier (
+        barrier: Readonly<GeneralBarrier> | null,
+        bufferBarriers?: readonly BufferBarrier[] | undefined,
+        buffers?: readonly Buffer[] | undefined,
+        textureBarriers?: readonly TextureBarrier[] | undefined,
+        textures?: readonly Texture[] | undefined,
+    ): void {
+        throw new Error('Method not implemented.');
+    }
+    public blitTexture (srcTexture: Readonly<Texture>, dstTexture: Texture, regions: readonly TextureBlit[], filter: Filter): void {
+        throw new Error('Method not implemented.');
+    }
+
+    public cmdPackage: WebGPUCmdPackage = new WebGPUCmdPackage();
+    protected _webGPUAllocator: WebGPUCommandAllocator | null = null;
+    protected _isInRenderPass = false;
+    protected _curGPUPipelineState: IWebGPUGPUPipelineState | null = null;
+    protected _curWebGPUPipelineState: WebGPUPipelineState | null = null;
+    protected _curGPUDescriptorSets: IWebGPUGPUDescriptorSet[] = [];
+    protected _curGPUInputAssembler: IWebGPUGPUInputAssembler | null = null;
+    protected _curDynamicOffsets: number[][] = [];
+    protected _curViewport: Viewport | null = null;
+    protected _curScissor: Rect | null = null;
+    protected _curLineWidth: number | null = null;
+    protected _curDepthBias: IWebGPUDepthBias | null = null;
+    protected _curBlendConstants: number[] = [];
+    protected _curDepthBounds: IWebGPUDepthBounds | null = null;
+    protected _curStencilWriteMask: IWebGPUStencilWriteMask | null = null;
+    protected _curStencilCompareMask: IWebGPUStencilCompareMask | null = null;
+    protected _isStateValid = false;
+    protected _globalDescriptors: WebGPUDescriptorSet[] = [];
```

Comment:
Why use protected? Will it be inherited?

Reply:
There are asynchronous versions of CommandBuffer, and some asynchronous operations might inherit from this class. Therefore, I'm using the protected access modifier for attribute definitions